python_modules/dagster/dagster_tests/definitions_tests/auto_materialize_tests/asset_daemon_scenario.py

```diff
@@ -0,0 +1,246 @@
+                materialize(
+                    assets=self.assets,
+                    instance=self.instance,
+                    partition_key=run_request.partition_key,
+                    tags=run_request.tags,
+                    raise_on_error=False,
+                    selection=run_request.asset_selection,
+                )
+        return self
+
+    def add_requested_runs(self) -> "AssetDaemonScenarioState":
+        return self.with_runs(*self.run_requests)
+
+    def evaluate_tick(self) -> "AssetDaemonScenarioState":
+        with pendulum.test(self.current_time):
+            new_run_requests, new_cursor, new_evaluations = AssetDaemonContext(
+                asset_graph=self.asset_graph,
+                target_asset_keys=None,
+                instance=self.instance,
+                materialize_run_tags={},
+                observe_run_tags={},
+                cursor=self.cursor,
+                auto_observe=True,
+                respect_materialization_data_versions=False,
+                logger=self.logger,
+            ).evaluate()
+        return self._replace(
+            run_requests=new_run_requests,
+            cursor=new_cursor,
+            evaluations=new_evaluations,
+        )
+
+    def _log_assertion_error(self, expected: Sequence[Any], actual: Sequence[Any]) -> None:
+        expected_str = "\n\n".join("\t" + str(rr) for rr in expected)
+        actual_str = "\n\n".join("\t" + str(rr) for rr in actual)
+        message = f"\nExpected: \n\n{expected_str}\n\nActual: \n\n{actual_str}\n"
+        self.logger.error(message)
+
+    def assert_requested_runs(
+        self, *expected_run_requests: RunRequest
+    ) -> "AssetDaemonScenarioState":
+        def sort_run_request_key_fn(run_request) -> Tuple[AssetKey, Optional[str]]:
+            return (min(run_request.asset_selection), run_request.partition_key)
+
+        sorted_run_requests = sorted(self.run_requests, key=sort_run_request_key_fn)
+        sorted_expected_run_requests = sorted(expected_run_requests, key=sort_run_request_key_fn)
+
+        try:
+            assert len(sorted_run_requests) == len(sorted_expected_run_requests)
+            for arr, err in zip(sorted_run_requests, sorted_expected_run_requests):
+                assert set(arr.asset_selection or []) == set(err.asset_selection or [])
+                assert arr.partition_key == err.partition_key
+        except:
+            self._log_assertion_error(sorted_expected_run_requests, sorted_run_requests)
+            raise
+
+        return self
+
+    def assert_evaluation(
+        self,
+        key: CoercibleToAssetKey,
+        expected_evaluation_specs: Sequence[AssetRuleEvaluationSpec],
+        num_requested: Optional[int] = None,
+        num_skipped: Optional[int] = None,
+        num_discarded: Optional[int] = None,
+    ) -> "AssetDaemonScenarioState":
+        """Makes an assertion about the evaluation data associated with a specific asset."""
```

Comment:
Looking at the name of this method and the docstring, I'm not able to immediately grasp what it's asserting. I think this would be easier to grok if phrased as "Assert that...".